### PR TITLE
[feat] Add Unit Tests to bothan-core

### PR DESCRIPTION
**Pull request by warittornc**
## Implementation details
- Do 1
- Do 2

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

**Review by RogerKSI — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]
ok, with this PR. However, I still see some active set logic left in the code. please recheck and remove it as well.

File: bothan-core/src/manager/crypto_asset_info/price/tasks.rs [hunk 1]
```diff
@@ -229,3 +218,343 @@ fn compute_source_routes(
...
+    fn test_compute_source_routes_with_unsupported_prerequisites() {
+        let routes = vec![
+            OperationRoute::new("A", Operation::Multiply),
+            OperationRoute::new("B", Operation::Multiply),
+            OperationRoute::new("C", Operation::Divide),
+        ];
+        let start = Decimal::one();
+        let mut cache = PriceCache::new();
+        cache.set_available("A".to_string(), Decimal::from(10710110));
+        cache.set_unavailable("B".to_string());
```
Comment (colmazia) [inline bothan-core/src/manager/crypto_asset_info/price/tasks.rs:549]: shouldn't it be unsupported?

File: bothan-core/src/registry/validate.rs [hunk 2]
```diff
@@ -50,3 +50,40 @@ pub(crate) fn validate_signal(
...
+
+        let res = validate_signal("CS:DNE-USD", &mut HashMap::new(), &registry);
+        assert_eq!(
+            res,
+            Err(ValidationError::InvalidDependency("CS:DNE-USD".to_string()))
+        );
+    }
+
+    #[test]
+    fn test_validate_signal_with_circular_dependency() {
```
Comment (colmazia) [inline bothan-core/src/registry/validate.rs:79]: can you add partial circular case

Comment (warittornc) [reply bothan-core/src/registry/validate.rs:79]: validate cannot distinguish between the two, I don't think its necessary

**Review by colmazia — APPROVED** [review]

**Pull request merged**